fdbserver/TLogServer.actor.cpp

```diff
@@ -2120,7 +2134,10 @@ Future<Void> tLogPeekMessages(PromiseType replyPromise,
 	auto messagesValue = messages.toValue();
 	reply.arena.dependsOn(messagesValue.arena());
 	reply.messages = messagesValue;
-	reply.end = clusterRecoveryVersion.present() ? clusterRecoveryVersion.get() : endVersion;
+	reply.end = endVersion;
+	if (clusterRecoveryVersion.present() && logData->stopped() && logData->version.get() < reqBegin) {
+		reply.end = clusterRecoveryVersion.get();
```

Comment:
It might be better add +1 here, instead of the place `clusterRecoveryVersion` was set.